app/services/work_packages/copy_service.rb

```diff
@@ -111,4 +112,29 @@ def copy_work_package_attachments(copy)
       references: %i[description]
     )
   end
+
+  def copy_share_members(copy, send_notifications)
+    work_package.members.each do |member|
+      create_share_membership(member, copy, send_notifications)
+    end
+  end
+
+  private
+
+  def create_share_membership(member, target, send_notifications)
+    role_ids = member.member_roles.map(&:role_id)
+
+    return if role_ids.empty?
+
+    attributes = member
+                   .attributes.dup.except("id", "project_id", "entity_id", "created_at", "updated_at")
+                   .merge(role_ids:,
+                          project: target.project,
+                          entity: target,
+                          send_notifications:)
+
+    Members::CreateService
```

Comment:
Based on my quick comparison between the two, there shouldn't be a difference currently between using the `Members::CreateService` and the `Shares::CreateService` but semantically, the later would be 'more correct'. It might also be that the two diverge more later so I would use the `Shares::CreateService` to be on the safe side.

Reply:
Good point, thanks!